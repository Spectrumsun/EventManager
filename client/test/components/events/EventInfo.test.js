import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import ConnectedEventInfo,
{ EventInfo, mapDispatchToProps } from '../../../src/components/Event/EventInfo';

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  events: {
    loadedEvent: {
      name: 'tom',
      startDate: '2018-10-20',
      endDate: '2018-10-23',
      time: '10:00',
      purpose: 'for test',
      center: 1,
    },
    error: false
  },
  match: {
    params: {
      id: '1'
    }
  },
  auth: {
    isAuthenticated: false,
    user: {}
  }
};

const store = mockStore(initialState);

const props = {
  onOneEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  onDeleteEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  page: {
    pages: 5
  },
  initPostEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  loadedEvent: {
    name: 'tom',
    startDate: '2018-10-20',
    endDate: '2018-10-23',
    time: '10:00',
    purpose: 'for test',
    center: 1,
  },
  error: false,
  match: {
    params: {
      id: '1'
    }
  },
  auth: {
    isAuthenticated: false,
    user: {}
  },
  history: createMemoryHistory()
};

const mountedWrapper = mount(<Provider store={store}>
  <BrowserRouter>
    <ConnectedEventInfo {...props} />
  </BrowserRouter>
</Provider>);

const shallowWrapper = shallow(<EventInfo {...props} />);

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'tomato',
    startDate: '2018-12-05',
    endDate: '2018-06-21',
    time: '21:00',
    purpose: 'fun',
    centerId: 1,
    totalPage: '1',
    next: 1,
    centerName: 'yaba center',
    pageNumber: '2',
    formValid: false,
  }
};

describe('<EventInfo /> Component', () => {
  const wrapper = shallow(<EventInfo {...props} />);
  it('should render the <EventInfo />', () => {
    shallow(<EventInfo {...props} />);
  });

  it('should render the <EventInfo /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('EventInfo').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedEventInfo {...props} />
        </BrowserRouter>
      </Provider>);
    expect(tree).toMatchSnapshot();
  });


  it('should render initial layout of <EventInfo />', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('calls deleteEvent event to delete an event when delete button is clicked', () => {
    sinon.spy(shallowWrapper.instance(), 'deleteEvent');
    shallowWrapper.instance().deleteEvent(event);
    expect(shallowWrapper.instance().deleteEvent.calledOnce).toEqual(true);
  });

  it('should have div element match snap', () => {
    expect(wrapper.getElements('div')).toMatchSnapshot();
  });

  it('should have div element', () => {
    expect(wrapper.find('div').length).toEqual(7);
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).onOneEvent).toBeTruthy();
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).onDeleteEvent).toBeTruthy();
  });
});

