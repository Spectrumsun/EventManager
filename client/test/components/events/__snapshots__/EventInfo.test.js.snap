// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventInfo /> Component should have div element match snap 1`] = `
Array [
  <div>
    <div>
      <div
        className="container"
        style={
          Object {
            "paddingTop": "100px",
          }
        }
      >
        <div
          className="container"
          style={
            Object {
              "width": "45rem",
            }
          }
        >
          <div
            className="card loginCard"
          >
            <div
              className="card-header dark"
            >
              <h1
                className="color"
              >
                Event Info
              </h1>
            </div>
            <div
              className="card-body"
            >
              <h5>
                <strong>
                  Event Name
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Start Date
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  End Date
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Time
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Purpose
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Center
                </strong>
              </h5>
              <ul
                className="list-group col-md-6"
              >
                <Link
                  replace={false}
                  style={
                    Object {
                      "color": "#35434A",
                    }
                  }
                  to="/centers/undefined"
                >
                  <h6>
                    <li
                      className="list-group-item centerlist"
                    />
                  </h6>
                </Link>
              </ul>
              <br />
            </div>
          </div>
        </div>
      </div>
      <footer />
    </div>
  </div>,
]
`;

exports[`<EventInfo /> Component should match component snapshot 1`] = `
<div>
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="container"
        style={
          Object {
            "width": "45rem",
          }
        }
      >
        <div
          className="card loginCard"
        >
          <div
            className="card-header dark"
          >
            <h1
              className="color"
            >
              Event Info
            </h1>
          </div>
          <div
            className="card-body"
          >
            <h5>
              <strong>
                Event Name
              </strong>
            </h5>
            <h6
              className="list-group-item"
            />
            <br />
            <h5>
              <strong>
                Start Date
              </strong>
            </h5>
            <h6
              className="list-group-item"
            >
              2018-10-20
            </h6>
            <br />
            <h5>
              <strong>
                End Date
              </strong>
            </h5>
            <h6
              className="list-group-item"
            >
              2018-10-23
            </h6>
            <br />
            <h5>
              <strong>
                Time
              </strong>
            </h5>
            <h6
              className="list-group-item"
            >
              10:00
            </h6>
            <br />
            <h5>
              <strong>
                Purpose
              </strong>
            </h5>
            <h6
              className="list-group-item"
            >
              for test
            </h6>
            <br />
            <h5>
              <strong>
                Center
              </strong>
            </h5>
            <ul
              className="list-group col-md-6"
            >
              <a
                href="/centers/undefined"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#35434A",
                  }
                }
              >
                <h6>
                  <li
                    className="list-group-item centerlist"
                  />
                </h6>
              </a>
            </ul>
            <br />
          </div>
        </div>
      </div>
    </div>
    <div
      className="center"
    >
      <footer>
        <div>
          <h3
            className="center"
          >
            Event Manager
          </h3>
        </div>
        <div
          style={
            Object {
              "display": "inline-block",
              "textAlign": "left",
            }
          }
        >
          <div>
            <div
              className="float-left"
            >
              <a
                className="navbar-brand"
                href="/"
              >
                <img
                  alt=""
                  height="30"
                  src="https://res.cloudinary.com/skybound/image/upload/v1522444986/eventmanager/static/event.png"
                  width="30"
                />
              </a>
            </div>
            <div
              className="float-left"
            >
              <ul>
                <p
                  className="footerTitle"
                >
                  Social Media
                </p>
                <li
                  className="footerColor"
                >
                  facebook
                </li>
                <li
                  className="footerColor"
                >
                  twitter
                </li>
                <li
                  className="footerColor"
                >
                  instargram
                </li>
              </ul>
            </div>
            <div
              className="float-left"
            >
              <ul>
                <p
                  className="footerTitle"
                >
                  Event manager
                </p>
                <li
                  className="footerColor"
                >
                  Questions
                </li>
                <li
                  className="footerColor"
                >
                  Jobs
                </li>
                <li
                  className="footerColor"
                >
                  Help
                </li>
              </ul>
            </div>
            <div
              className="float-left"
            >
              <ul>
                <p
                  className="footerTitle"
                >
                  About
                </p>
                <li
                  className="footerColor"
                >
                  desgin by Taiwo Sunday
                </li>
                <li
                  className="footerColor"
                >
                  +2348038512455
                </li>
                <li
                  className="footerColor"
                >
                  https://github.com/Spectrumsun
                </li>
                <li
                  className="footerColor"
                >
                  taiwo.sunday@outlook.com
                </li>
              </ul>
            </div>
            <div
              className="float-left"
            >
              <ul>
                <p
                  className="footerTitle"
                >
                  Site
                </p>
                <li
                  className="footerColor"
                >
                  Home
                </li>
                <li
                  className="footerColor"
                >
                  Centers
                </li>
                <li
                  className="footerColor"
                >
                  Events
                </li>
              </ul>
            </div>
          </div>
        </div>
        <p
          className="center"
        >
          Event Managment Â© 2017
        </p>
      </footer>
    </div>
  </div>
</div>
`;

exports[`<EventInfo /> Component should render initial layout of <EventInfo /> 1`] = `
Array [
  <div>
    <div>
      <div
        className="container"
        style={
          Object {
            "paddingTop": "100px",
          }
        }
      >
        <div
          className="container"
          style={
            Object {
              "width": "45rem",
            }
          }
        >
          <div
            className="card loginCard"
          >
            <div
              className="card-header dark"
            >
              <h1
                className="color"
              >
                Event Info
              </h1>
            </div>
            <div
              className="card-body"
            >
              <h5>
                <strong>
                  Event Name
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Start Date
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  End Date
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Time
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Purpose
                </strong>
              </h5>
              <h6
                className="list-group-item"
              />
              <br />
              <h5>
                <strong>
                  Center
                </strong>
              </h5>
              <ul
                className="list-group col-md-6"
              >
                <Link
                  replace={false}
                  style={
                    Object {
                      "color": "#35434A",
                    }
                  }
                  to="/centers/undefined"
                >
                  <h6>
                    <li
                      className="list-group-item centerlist"
                    />
                  </h6>
                </Link>
              </ul>
              <br />
            </div>
          </div>
        </div>
      </div>
      <footer />
    </div>
  </div>,
]
`;
